compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml

```diff
@@ -196,16 +196,21 @@ class HasFromAnyContractKey t k | t -> k where
   -- | HIDE
   _fromAnyContractKey : proxy t -> Any -> Optional k
 
--- | Exercise a choice on the contract associated with the given key.
---
--- You must pass the `t` using an explicit type application. For
--- instance, if you want to exercise a choice `Withdraw` on a contract of
--- template `Account` given by its key `k`, you must call
--- `exerciseByKey @Account k Withdraw`.
-exerciseByKey : forall t k c r. (HasFetchByKey t k, HasExercise t c r) => k -> c -> Update r
-exerciseByKey k c = do
-    (cid, _) <- fetchByKey @t k
-    exercise cid c
+-- | Constraint satisfied by choices from templates with key.
+type ChoiceByKey t c k r =
```

Comment:
not sure if this was intentional but the arguments are transposed here wrt `HasExerciseByKey` below
`t c k r`
`t k c r`
we probably want to keep the latter?